builder/lxc/step_export.go

```diff
@@ -84,9 +83,37 @@ func (s *stepExport) Run(state multistep.StateBag) multistep.StepAction {
 	commands[0] = []string{
 		"lxc-stop", "--name", name,
 	}
-	commands[1] = []string{
-		"tar", "-C", containerDir, "--numeric-owner", "--anchored", "--exclude=./rootfs/dev/log", "-czf", outputPath, "./rootfs",
+
+	if len(config.ExportConfig.Folders) == 0 {
+		commands[1] = []string{
+			"tar", "-C", containerDir, "--numeric-owner", "--anchored", "--exclude=./rootfs/dev/log", "-czf", "rootfs.tar.gz", "./rootfs",
+		}
+	} else {
+		ui.Say("Preparing container to export...")
+		filename := "rootfs.tar.gz"
+		if config.ExportConfig.Filename != "" {
+			filename = config.ExportConfig.Filename
+		}
+		outputPath := filepath.Join(config.OutputDir, filename)
+		err, exportFolder := s.PrepareExport(containerDir, config.ExportConfig.Folders)
+		if err != nil {
+			err := fmt.Errorf("Error creating container export folder: %s", err)
+			state.Put("error", err)
+			ui.Error(err.Error())
+			return multistep.ActionHalt
+		}
+		command := []string{
+			"tar", "-C", exportFolder, "--anchored", "-czf", outputPath, ".",
+		}
+		if config.ExportConfig.Owner != "" {
+			command = append(command, []string{"--owner", config.ExportConfig.Owner}...)
```

Comment:
OK, now I'm even more confused! `tar` is supposed to preserve permissions by default during archive, isn't it? It is also supposed to preserve permissions during extraction when run as the super user. Where are we running into issues exactly then?

This feels like a workaround for a problem we maybe don't fully understand?